File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/log_experiment_run_batch_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class LogExperimentRunBatchDetails(object):
    """
    Run Batch data to log.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new LogExperimentRunBatchDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param run_id:
            The value to assign to the run_id property of this LogExperimentRunBatchDetails.
        :type run_id: str

        :param metrics:
            The value to assign to the metrics property of this LogExperimentRunBatchDetails.
        :type metrics: list[oci.aidataplatform_dp.models.ExperimentRunMetric]

        :param params:
            The value to assign to the params property of this LogExperimentRunBatchDetails.
        :type params: list[oci.aidataplatform_dp.models.ExperimentRunParam]

        :param tags:
            The value to assign to the tags property of this LogExperimentRunBatchDetails.
        :type tags: list[oci.aidataplatform_dp.models.ExperimentRunTag]

        """
        self.swagger_types = {
            'run_id': 'str',
            'metrics': 'list[ExperimentRunMetric]',
            'params': 'list[ExperimentRunParam]',
            'tags': 'list[ExperimentRunTag]'
        }

        self.attribute_map = {
            'run_id': 'run_id',
            'metrics': 'metrics',
            'params': 'params',
            'tags': 'tags'
        }

        self._run_id = None
        self._metrics = None
        self._params = None
        self._tags = None

    @property
    def run_id(self):
        """
        **[Required]** Gets the run_id of this LogExperimentRunBatchDetails.
        Unique identifier for the run.


        :return: The run_id of this LogExperimentRunBatchDetails.
        :rtype: str
        """
        return self._run_id

    @run_id.setter
    def run_id(self, run_id):
        """
        Sets the run_id of this LogExperimentRunBatchDetails.
        Unique identifier for the run.


        :param run_id: The run_id of this LogExperimentRunBatchDetails.
        :type: str
        """
        self._run_id = run_id

    @property
    def metrics(self):
        """
        Gets the metrics of this LogExperimentRunBatchDetails.
        Metrics to log.


        :return: The metrics of this LogExperimentRunBatchDetails.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentRunMetric]
        """
        return self._metrics

    @metrics.setter
    def metrics(self, metrics):
        """
        Sets the metrics of this LogExperimentRunBatchDetails.
        Metrics to log.


        :param metrics: The metrics of this LogExperimentRunBatchDetails.
        :type: list[oci.aidataplatform_dp.models.ExperimentRunMetric]
        """
        self._metrics = metrics

    @property
    def params(self):
        """
        Gets the params of this LogExperimentRunBatchDetails.
        Params to log.


        :return: The params of this LogExperimentRunBatchDetails.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentRunParam]
        """
        return self._params

    @params.setter
    def params(self, params):
        """
        Sets the params of this LogExperimentRunBatchDetails.
        Params to log.


        :param params: The params of this LogExperimentRunBatchDetails.
        :type: list[oci.aidataplatform_dp.models.ExperimentRunParam]
        """
        self._params = params

    @property
    def tags(self):
        """
        Gets the tags of this LogExperimentRunBatchDetails.
        Tags to log.


        :return: The tags of this LogExperimentRunBatchDetails.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentRunTag]
        """
        return self._tags

    @tags.setter
    def tags(self, tags):
        """
        Sets the tags of this LogExperimentRunBatchDetails.
        Tags to log.


        :param tags: The tags of this LogExperimentRunBatchDetails.
        :type: list[oci.aidataplatform_dp.models.ExperimentRunTag]
        """
        self._tags = tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/LogExperimentRunBatchDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Run batch data to log.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=LogExperimentRunBatchDetails.Builder.class)

public final class LogExperimentRunBatchDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"runId", "metrics", "params", "tags"})
    public LogExperimentRunBatchDetails(String runId, java.util.List<ExperimentRunMetric> metrics, java.util.List<ExperimentRunParam> params, java.util.List<ExperimentRunTag> tags) {
        super();
        this.runId = runId;
        this.metrics = metrics;
        this.params = params;
        this.tags = tags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Unique identifier for the run.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("run_id")
private String runId;

        /**
         * Unique identifier for the run.
         * @param runId the value to set
         * @return this builder
         **/
        

public Builder runId(String runId) {
    this.runId = runId;
    return this;
}
            /**
     * Metrics to log.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("metrics")
private java.util.List<ExperimentRunMetric> metrics;

        /**
         * Metrics to log.
         * @param metrics the value to set
         * @return this builder
         **/
        

public Builder metrics(java.util.List<ExperimentRunMetric> metrics) {
    this.metrics = metrics;
    return this;
}
            /**
     * Params to log.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("params")
private java.util.List<ExperimentRunParam> params;

        /**
         * Params to log.
         * @param params the value to set
         * @return this builder
         **/
        

public Builder params(java.util.List<ExperimentRunParam> params) {
    this.params = params;
    return this;
}
            /**
     * Tags to log.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("tags")
private java.util.List<ExperimentRunTag> tags;

        /**
         * Tags to log.
         * @param tags the value to set
         * @return this builder
         **/
        

public Builder tags(java.util.List<ExperimentRunTag> tags) {
    this.tags = tags;
    return this;
}


        public LogExperimentRunBatchDetails build() {
            LogExperimentRunBatchDetails model = new LogExperimentRunBatchDetails(this.runId
                , this.metrics
                , this.params
                , this.tags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(LogExperimentRunBatchDetails model) {
                this.runId(model.getRunId());
    this.metrics(model.getMetrics());
    this.params(model.getParams());
    this.tags(model.getTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique identifier for the run.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("run_id")
    private final String runId;

        /**
     * Unique identifier for the run.
     * @return the value
     **/
    
    public String getRunId() {
        return runId;
    }


        /**
     * Metrics to log.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("metrics")
    private final java.util.List<ExperimentRunMetric> metrics;

        /**
     * Metrics to log.
     * @return the value
     **/
    
    public java.util.List<ExperimentRunMetric> getMetrics() {
        return metrics;
    }


        /**
     * Params to log.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("params")
    private final java.util.List<ExperimentRunParam> params;

        /**
     * Params to log.
     * @return the value
     **/
    
    public java.util.List<ExperimentRunParam> getParams() {
        return params;
    }


        /**
     * Tags to log.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tags")
    private final java.util.List<ExperimentRunTag> tags;

        /**
     * Tags to log.
     * @return the value
     **/
    
    public java.util.List<ExperimentRunTag> getTags() {
        return tags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("LogExperimentRunBatchDetails(");
        sb.append("runId=").append(String.valueOf(this.runId));
        sb.append(", metrics=").append(String.valueOf(this.metrics));
        sb.append(", params=").append(String.valueOf(this.params));
        sb.append(", tags=").append(String.valueOf(this.tags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof LogExperimentRunBatchDetails)) {
            return false;
        }

        LogExperimentRunBatchDetails other = (LogExperimentRunBatchDetails) o;
        return java.util.Objects.equals(this.runId, other.runId) &&
            java.util.Objects.equals(this.metrics, other.metrics) &&
            java.util.Objects.equals(this.params, other.params) &&
            java.util.Objects.equals(this.tags, other.tags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.runId == null ? 43 : this.runId.hashCode());
        result = (result * PRIME) + (this.metrics == null ? 43 : this.metrics.hashCode());
        result = (result * PRIME) + (this.params == null ? 43 : this.params.hashCode());
        result = (result * PRIME) + (this.tags == null ? 43 : this.tags.hashCode());
        return result;
    }


}
